src/tree.c

```diff
@@ -641,7 +643,12 @@ int git_tree__write_index(
 		git_index__set_ignore_case(index, false);
 	}
 
-	ret = write_tree(oid, repo, index, "", 0);
+	/* Grow the buffer beforehand to 1kb */
+	if ((ret = git_buf_grow(&shared_buf, 1024)) >= 0)
+	{
+		ret = write_tree(oid, repo, index, "", 0, &shared_buf);
+		git_buf_free(&shared_buf);
```

Comment:
The free doesn't need to be conditional, it'll do the right thing in case of OOM.